packages/skygear-core/lib/user.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright 2015 Oursky Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import _ from 'lodash';
+import {toJSON, fromJSON} from './util';
+
+const _metaAttrs = {
+  user_id: { //eslint-disable-line
+    parser: (v) => {
+      return v;
+    },
+    stringify: (v) => {
+      return v;
+    },
+    newKey: 'userID'
+  },
+  created_at: { //eslint-disable-line
+    parser: (v) => {
+      return new Date(v);
+    },
+    stringify: (v) => {
+      return v.toJSON();
+    },
+    newKey: 'createdAt'
+  },
+  updated_at: { //eslint-disable-line
+    parser: (v) => {
+      return new Date(v);
+    },
+    stringify: (v) => {
+      return v.toJSON();
+    },
+    newKey: 'updatedAt'
+  },
+  last_login_at: { //eslint-disable-line
+    parser: (v) => {
+      return new Date(v);
+    },
+    stringify: (v) => {
+      return v.toJSON();
+    },
+    newKey: 'lastLoginAt'
+  },
+  last_seen_at: { //eslint-disable-line
+    parser: (v) => {
+      return new Date(v);
+    },
+    stringify: (v) => {
+      return v.toJSON();
+    },
+    newKey: 'lastSeenAt'
+  },
+  created_by: { //eslint-disable-line
+    parser: (v) => {
+      return v;
+    },
+    stringify: (v) => {
+      return v;
+    },
+    newKey: 'createdBy'
+  },
+  updated_by: { //eslint-disable-line
+    parser: (v) => {
+      return v;
+    },
+    stringify: (v) => {
+      return v;
+    },
+    newKey: 'updatedBy'
+  }
+};
+
+const _metaKeys = _.reduce(_metaAttrs, function (result, value, key) {
+  result[value.newKey] = key;
+  return result;
+}, {});
+
+/**
+ * User provides the model for Skygear User.
+ *
+ */
+export default class User {
+
+  constructor(attrs) {
+    this.update(attrs);
+  }
+
+  /**
+   * Gets all keys of attributes of the records. Skygear reserved keys, that is
+   * underscore prefixed keys, are excluded.
+   *
+   * @type {String[]}
+   */
+  get attributeKeys() {
```

Comment:
The user object put all dynamic fields inside the `metadata`, the object itself is static, so this function can be removed.

Same for the function `update`, which can be simpler I think.

Reply:
@Steven-Chan Updated.